src/main/java/uk/gov/hmcts/reform/sscs/validators/SscsCaseValidator.java

```diff
@@ -296,16 +300,31 @@ private void checkPersonAddressAndDob(Address address, Identity identity, String
 
         if (!doesAddressLine1Exist(address)) {
             warnings.add(getMessageByCallbackType(callbackType, personType, getWarningMessageName(personType, appellant) + ADDRESS_LINE1, IS_EMPTY));
+        } else {
+            if (!address.getLine1().matches(ADDRESS_REGEX)) {
+                warnings.add(getMessageByCallbackType(callbackType, personType, getWarningMessageName(personType, appellant) + ADDRESS_LINE1, HAS_INVALID_ADDRESS));
+            }
         }
+
+        String townLine = (isAddressLine4Present) ? ADDRESS_LINE3 : ADDRESS_LINE2;
         if (!doesAddressTownExist(address)) {
-            String addressLine = (isAddressLine4Present) ? ADDRESS_LINE3 : ADDRESS_LINE2;
-            warnings.add(getMessageByCallbackType(callbackType, personType, getWarningMessageName(personType, appellant) + addressLine, IS_EMPTY));
 
+            warnings.add(getMessageByCallbackType(callbackType, personType, getWarningMessageName(personType, appellant) + townLine, IS_EMPTY));
+        } else {
+            if (!address.getTown().matches(ADDRESS_REGEX)) {
```

Comment:
Does this ADDRESS_REGEX ignore blank value?

Reply:
No, minimum of 2 characters

Reply:
We have optional address field line 2 or line 3 which may be the issue here. as this check is in else condition of if(!doesAddressTownExist(address))